### PR TITLE
split platform builds

### PR DESCRIPTION
**Pull request by gorkem**
Splits the release builds to corresponding platforms. Introduces go-releaser configuration for each platform. adds a new workflow that uses GH CLI to do the publish. 

related to #196

### CODE REVIEW

**Review by amisevsk — APPROVED** [review]
LGTM, sorry I missed this for a couple days.

Minor request: some files have no trailing newline, others have multiple

File: .github/workflows/platform-release.yaml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  release:
+    runs-on: ubuntu-latest
+    needs: [build-linux, build-macos, build-windows]
+    steps:
+      - name: Merge built artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: dist
+          pattern: dist-*
+          merge-multiple: true
```
Comment (amisevsk) [inline .github/workflows/platform-release.yaml:132]: I'm not sure how artifacts work -- if we run two releases within a week of each other will they see both artifacts? I suppose the second build would overwrite them?

Comment (gorkem) [reply .github/workflows/platform-release.yaml:132]: You can only download artifacts that were uploaded during the same workflow run or use the UI to navigate to the workflow run and download. I have also done a shortened retention, the default is 90 days.

Comment (amisevsk) [reply .github/workflows/platform-release.yaml:132]: That makes sense :+1:

**Comment by gorkem** [comment]
> LGTM, sorry I missed this for a couple days.
> 
> Minor request: some files have no trailing newline, others have multiple

fixed

**Pull request merged**